MyWebEngineView.h

```diff
@@ -66,4 +78,45 @@ class MyWebEngineView : public QWebEngineView {
     // engine to use the current view for the new window's contents.
     return this;
   }
+
+private:
+  /**
+   * @brief Handles the translate action from the context menu.
+   *
+   * If text is selected, opens Google Translate with the selected text.
+   * Otherwise, opens Google Translate with the current page URL for
+   * full page translation. Uses the system default browser to open
+   * the translation URL.
+   */
+  void handleTranslateAction() {
+    auto page = this->page();
+    if (!page) return;
+
+    QString selectedText = page->selectedText();
+    QUrl translateUrl;
+
+    if (!selectedText.isEmpty()) {
+      // Translate selected text
+      QUrlQuery query;
+      query.addQueryItem("text", selectedText);
+      query.addQueryItem("op", "translate");
+      translateUrl.setUrl("https://translate.google.com/");
+      translateUrl.setQuery(query);
+    } else {
+      // Translate entire page
+      QString currentUrl = page->url().toString();
+      if (!currentUrl.isEmpty()) {
+        QUrlQuery query;
+        query.addQueryItem("u", currentUrl);
+        query.addQueryItem("sl", "auto");
+        query.addQueryItem("tl", "en");
+        translateUrl.setUrl("https://translate.google.com/translate");
+        translateUrl.setQuery(query);
+      }
+    }
+
+    if (translateUrl.isValid()) {
+      QDesktopServices::openUrl(translateUrl);
```

Comment:
lol! This app, Phraim, is a web browser!
Have this app open a new SplitterWindow to the translateUrl.